### PR TITLE
[nr-k8s-otel-collector] - exposing pod EVs to configuration in values file

### PR DESCRIPTION
**Pull request by GHKhuddle1**
#### Is this a new chart
No. This is a modification to 'nr-otel-for-k8s', which is currently maintained by @csongnr and @dbudziwojskiNR 

#### What this PR does / why we need it:
This PR exposes the option to add additional pod EVs to the deployment and daemonset running on kubernetes. I ran into a use case where I needed to be able to customize the 'nr-otel-for-k8s' service with a custom collector config, but once I did so I was unable to securely pass values to that config. In order to securely pass values to a custom config, having access from the values file to provide additional EVs is helpful.

#### Checklist
- [x] Chart Version bumped
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[mychartname]`)

### CODE REVIEW

File: charts/nr-k8s-otel-collector/templates/deployment.yaml [hunk 1]
```diff
@@ -46,6 +46,10 @@ spec:
           args: ["--config", "/config/deployment-config.yaml"]
           resources:
             {{- toYaml .Values.deployment.resources | nindent 12 }}
+          {{- with .Values.deployment.envsFrom }}
+          envFrom:
+          {{- . | toYaml | nindent 12 }}
```
Comment (dbudziwojskiNR) [inline charts/nr-k8s-otel-collector/templates/deployment.yaml:51]: Please indent this line

File: charts/nr-k8s-otel-collector/templates/daemonset.yaml [hunk 2]
```diff
@@ -44,6 +44,10 @@ spec:
           args: ["--config", "/config/daemonset-config.yaml"]
           resources:
             {{- toYaml .Values.daemonset.resources | nindent 12 }}
+          {{- with .Values.daemonset.envsFrom }}
+          envFrom:
+          {{- . | toYaml | nindent 12 }}
```
Comment (dbudziwojskiNR) [inline charts/nr-k8s-otel-collector/templates/daemonset.yaml:49]: Please indent this line

**Comment by GHKhuddle1** [comment]
> Thanks for your PR! Would you please create tests for this new functionality? You can see examples of other tests in `tests` folder. Thank you!

I am happy to. I have not used the helm unit testing framework before, but is it safe to say you are using the 'helm-unittest' plugin found here 'https://github.com/helm-unittest/helm-unittest?tab=readme-ov-file'?

I was attempting to run the plugin locally against the tests you already have defined and it appears to fail because it is unable to pull / access your New Relic common library.

`error calling include: template: no template "nrKubernetesOtel.deployment.fullname" associated with template "gotpl"`

Is there a way to get around this when running the unittesting framework locally? Or would you just copy the contents of the common library and dump it in the local testing directory?

**Comment by dbudziwojskiNR** [comment]
> I am happy to. I have not used the helm unit testing framework before, but is it safe to say you are using the 'helm-unittest' plugin found here 'https://github.com/helm-unittest/helm-unittest?tab=readme-ov-file'?

Thank you! Yes, you have the right link. The common library is [publicly accessible](https://github.com/newrelic/helm-charts/tree/master/library/common-library). By chance have you run `helm dependency update` on the chart? It should install the subchart locally allowing the unit tests to run. Please let me know if you're still having issues.

**Comment by GHKhuddle1** [comment]
> > I am happy to. I have not used the helm unit testing framework before, but is it safe to say you are using the 'helm-unittest' plugin found here 'https://github.com/helm-unittest/helm-unittest?tab=readme-ov-file'?
> 
> Thank you! Yes, you have the right link. The common library is [publicly accessible](https://github.com/newrelic/helm-charts/tree/master/library/common-library). By chance have you run `helm dependency update` on the chart? It should install the subchart locally allowing the unit tests to run. Please let me know if you're still having issues.

Done. I have made the necessary whitespace changes and created additional testing files for the new functionality. Here is a screenshot of a testing run that passed all tests, including the new ones 'envVar' and 'envVarFrom'. Please let me know if you have any other concerns.

**Comment by GHKhuddle1** [comment]
@dbudziwojskiNR Any update on this?

**Comment by dbudziwojskiNR** [comment]
> @dbudziwojskiNR Any update on this?

Thanks for the tests! Looking into this shortly.

**Review by dbudziwojskiNR — APPROVED** [review]

**Comment by dbudziwojskiNR** [comment]
@GHKhuddle1 Thank you for your contribution!

**Pull request merged**